### ISSUE TITLE
Mod doesn't show up in profiler view unless it has the `on_tick` event registered

### ISSUE THREAD

**Issue by stevenwdv** [justarandomgeek/vscode-factoriomod-debug#60]
It seems that a mod doesn't show up in the profiler flame panel if it has not registered an 'on tick' handler, even if it has an empty function.

Compare:

Without `on_click`, I don't see my mod:
![without on_click](https://user-images.githubusercontent.com/27223270/169650525-de18afae-a777-4ed8-9fce-d8915a3a6f5e.png)

With `on_click`, ah there it is!:
![with on_tick](https://user-images.githubusercontent.com/27223270/169650460-fcef014d-4c01-42ff-87fb-caa640428acf.png)

In both cases I'm selecting a number of large areas with my selection tool.

**Comment by justarandomgeek** [comment]
It doesn't report until it accumulates `profileUpdateRate` of hits per mod. Without the on_tick you're probably not hitting it enough. You can `remote.call("profiler","dump")` to make it dump everything it's got collected but not dumped yet immediately.

**Comment by stevenwdv** [comment]
Hm, so is that the number of times it calls a certain function? Or is it the number of times it sampled some line in the mod? Because if it's about execution time then it should find my mod, as for a large selection it would normally take a couple of seconds (with processing code not shown here), but still does not show up.

**Comment by justarandomgeek** [comment]
the number of times it makes a top level entry into that mod's lua

**Comment by stevenwdv** [comment]
Ah okay I guess that is the problem then

**Comment by stevenwdv** [comment]
Would it be possible to also take the execution time of each mod into account?

**Comment by justarandomgeek** [comment]
not really, no. But you can just adjust `profileUpdateRate` in your launch config
